### PR TITLE
Cell Pokemon Support

### PR DESCRIPTION
**Pull request by SkOODaT**
## Description
Adds Basic Cell Pokemon Support, It Could Be Expanded Possibly.
No IVs, No GPS, Their The "Grass" Pokemon
Mons Should Update If Seen At A Stop Or IV-ed

Thell Record Into DB As NULL Stop ID and NULL Spawn ID

Frontend Support, Tips Show Scancell Pokemon, Nearby Pokemon, Wild Pokemon

Not yet added ~ WIP - Pokemon Types To Database(Will Require DB Migration)

`/misc/grass.png`
grass

## Motivation and Context
They Have Been There, In The Data, Why Not Show Them, Rurals Areas Could Benifit By This Type Of Scanning Alot....

## How Has This Been Tested?
Locally On My Own Map, With All Devices Going.

## Screenshots (if appropriate):
CELL

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by wrinklytesties** [comment]
My personal take on this PR, esp since these specific wild pokemon were discussed during the early RDM days (when plus plus first dropped mapper functions), and as a result of those discussion, these specific mons in the GMO were excluded for the following reasons:

- Pokes are anywhere up to 200m range of the center point of the jump
- There is no unique identifiable location data, requiring far too many jumps to discover the location
- There is no unique anchor (pokestop) that reduces the range to a less resource and time heavy task

With that said while low device coverage and rural areas could benefit heavily from discovery, if these find their way on the front end display, there will be confusion amongst users as where the actual poke is located.  With nearby pokemon anchored to pokestops, it is within a 70m distance, making discovery easy.

At 200m range with no anchor, you'd have to tie this into the center point of whatever jump first returned it.  For users, that's a 200m range of a location to discover what is there.  If a custom IV priority method is developed to IV these mons with no location, that's up to 6-8 jumps. 

For large setups, this will make the DB incredibly busy.   If this were to be merged, it would be nice if you can make this configurable in the YML file, allowing the host to enable/disable it on docker start up.  There is going to be so much noise from this PR for incredibly large setups.

**Comment by SkOODaT** [comment]
well for a large setup a toggle could be added but this data is in the data being processed already its just not added to the database, as for users finding them, ive been using a pokemon type scheme on my front end since the beginning, i plan to possibly add to this PR... ie 
Untidadadtled

**Comment by wrinklytesties** [comment]
Yes, this will add additional inserts and updates.  So if you can make this configurable in the YML to be enabled/disabled, it will allow those with large setups to avoid this influx where the benefits are incredibly low for those type of setups.

Adding the configurable option, will make for easier performance testing as well.

File: resources/webroot/index.js.mustache [hunk 1]
```diff
@@ -1842,21 +1842,37 @@ function addPokemon(pokemon, ts) {
   })
 
   if (oldPokemon === undefined || pokemon.is_start) {
+    var latOffset = Math.random() * 0.0003 - 0.0001
+    var lonOffset = Math.random() * 0.0003 - 0.0001
+
     if (pokemon.pokestop_id !== null && pokemon.spawn_id === null) {
```
Comment (123FLO321) [inline resources/webroot/index.js.mustache:1848]: wouldn't ` pokemon.spawn_id === null` cover both cases here?
Is there a reason for duplicating the code here?

File: Sources/RealDeviceMap/Modell/Pokemon.swift [hunk 2]
```diff
@@ -277,48 +278,56 @@ class Pokemon: JSONConvertibleObject, WebHookEvent, Equatable, CustomStringConve
...
+                    """
+
+                guard let mysql = mysql ?? DBController.global.mysql else {
+                    Log.error(message: "[POKEMON] Failed to connect to database.")
+                    throw DBController.DBError()
+                }
 
-            let mysqlStmt = MySQLStmt(mysql)
-            _ = mysqlStmt.prepare(statement: sql)
+                let mysqlStmt = MySQLStmt(mysql)
```
Comment (123FLO321) [inline Sources/RealDeviceMap/Modell/Pokemon.swift:304]: why did the indentation change here and for the following lines?

**Review by 123FLO321 — CHANGES_REQUESTED** [review]
looks good, except some minor changes

**Comment by 123FLO321** [comment]
Also those pokemon should probably be excluded from IV checks because the exact location is hard to know:
(check if pokestopId and spawnId is null)
https://github.com/RealDeviceMap/RealDeviceMap/blob/6d5cab63ec629fc741d8c1b358cae264dfe22ff5/Sources/RealDeviceMap/Controller/InstanceControllers/IVInstanceController.swift#L170

**Comment by 123FLO321** [comment]
From my testing there are ~20% more spawns when using this PR.

**Comment by 123FLO321** [comment]
> Yes, this will add additional inserts and updates. So if you can make this configurable in the YML to be enabled/disabled, it will allow those with large setups to avoid this influx where the benefits are incredibly low for those type of setups.
> 
> Adding the configurable option, will make for easier performance testing as well.

Implemented via `NO_CELL_POKEMON` in 6e48209bcca0e50107840945f906c96a0e287257

**Pull request merged**